src/transformers/modeling_tf_utils.py

```diff
@@ -398,7 +398,7 @@ def run_call_with_unpacked_inputs(self, *args, **kwargs):
         fn_args_and_kwargs.update(dict(zip(func.__code__.co_varnames[1:], args)))
 
         # Encoder Decoder models delegate the application of the configuration options to their inner models.
-        if "encoder_decoder" in str(self).lower():
+        if "EncoderDecoder" in type(self).__name__:
```

Comment:
```suggestion
        if "EncoderDecoder" in self.__class__.__name__:
```
Nit, we usually use this for the class names in other places of the lib.